### DISCUSSION TITLE
Is there support for mamba-based models, or am I doing something wrong?

### DISCUSSION THREAD

**Discussion by zvxayr** [guidance-ai/guidance#790] [Q&A]
When using this [model](https://huggingface.co/devingulliver/mamba-gguf/tree/main), using "+" behaves as if I did "+=". The data bleeds even when I did not store the state into a new variable as shown below.

![image](https://github.com/guidance-ai/guidance/assets/20508904/eb89db6a-1e90-4bb2-a3bf-3f60b1f13647)

Is there anything wrong with the code? Is there a correct model to use than `LlamaCpp`?

**Comment by riedgar-ms** [comment]
That looks like the right way to load the model. I take it you were expecting the second cell to come back with "What passage, that's the first thing you've said to me" (or something similar)?

**Reply by zvxayr** [reply]
Yes. I couldn't regenerate answers with mamba models. I would need to reinitialize the model if I wanted to clear the memory.

**Reply by riedgar-ms** [reply]
@Harsha-Nori @paulbkoch would this be due to the caching the models do internally?
